imblearn/utils/_validation.py

```diff
@@ -99,10 +126,12 @@ def check_neighbors_object(nn_name, nn_object, additional_neighbor=0):
 
 def _count_class_sample(y):
     unique, counts = np.unique(y, return_counts=True)
+    if is_dask_container(unique):
+        unique, counts = unique.compute(), counts.compute()
```

Comment:
As written, this will fully execute the task graph of `y` twice. Once to compute `unique` and once to compute `counts`.

```suggestion
    if is_dask_container(unique):
        unique, counts = dask.compute(unique, counts)
```

You'll need to `import dask`